cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -309,6 +312,38 @@ public void testMapreduceHistory() throws Exception {
   }
 
   /**
+   * Tests history of a non existing program
+   */
+  @Test
+  public void testNonExistingProgramHistory() throws Exception {
+
+    Id.Program program = Id.Program.from(TEST_NAMESPACE2, DUMMY_APP_ID, ProgramType.MAPREDUCE, DUMMY_MR_NAME);
```

Comment:
`Id.Program` is deprecated. Use `ProgramId` instead. You can use `ProgramId.toId()` to convert to the old `Id.Program` when needed.